tools/lib/provision.py

```diff
@@ -247,12 +247,26 @@ def main(options):
         run(["sudo", "service", "redis-server", "restart"])
         run(["sudo", "service", "memcached", "restart"])
     if not options.is_production_travis:
-        # These won't be used anyway
-        run(["scripts/setup/configure-rabbitmq"])
-        # Need to set up Django before using is_template_database_current.
+
+        # Need to set up Django before using is_template_database_current and rabbitmq credentials
         os.environ.setdefault("DJANGO_SETTINGS_MODULE", "zproject.settings")
         import django
         django.setup()
+
+        import pika
+        from django.conf import settings
+        # These won't be used anyway
+        # check if rabbitmq is already configured
+        try:
+            credentials = pika.PlainCredentials(settings.RABBITMQ_USERNAME,
+                                                settings.RABBITMQ_PASSWORD)
+            pika.BlockingConnection(pika.ConnectionParameters(settings.RABBITMQ_HOST,
+                                                              credentials=credentials))
+            print("Rabbitmq is already configured.")
+
+        except pika.exceptions.ProbableAuthenticationError:
+            run(["scripts/setup/configure-rabbitmq"])
```

Comment:
I'm not a fan of this structure where the main work is happening inside an exception handler. Also that we spend several lines of quite gritty detail deciding whether to do something here, followed by just the one line to actually do it -- this script is in general pretty high-level, making one-line invocations of other scripts to do a lot of work, and the `pika.BlockingConnection` stuff is at a far more detailed level.

I think what I'd really like to see is a structure parallel to the DB init logic just below: something like
```
if options.is_force or is_rabbitmq_configured():
  run(["scripts/setup/configure-rabbitmq"])
else:
  print("RabbitMQ is already configured.")
```
Then you'd pull the rest of this logic into a function `is_rabbitmq_configured`.